### PR TITLE
Alpha 1 of test server

### PR DESCRIPTION
**Pull request by brywang-msft**
(Resubmitting - major change from last time is removal of dynamic compilation and using dotnet CLI to compile instead)
- ConsoleServer can login using Azure credentials, execute commands, and output response
- ConsoleServer uses stdin for JSON-RPC input, stdout for JSON-RPC output, and named pipes for logging
- Some code best practice cleanup work required, but we can worry about that later
- IO stuff was split into an IO library as there are future plans to use a generic JSON-RPC framework that should replace the IO library
- No test harnesses or integration tests added yet (still need to try integration with Azure SDK tests)
- 84/84 unit tests passing, also tested manually input/output operation of server using a test harness

### CODE REVIEW

File: PSSwagger.LiveTestFramework/build/PSSwagger.LiveTestFramework.Build.psd1 [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
+    GUID = '03459d8d-5fcc-48e1-9a88-94f7971e0335'
     Author = 'Microsoft Corporation'
     CompanyName = 'Microsoft Corporation'
     Copyright = '(c) Microsoft Corporation. All rights reserved.'
-    Description = 'PowerShell module with commands for generating or manipulating PSSwagger.LiveTestFramework binaries.'
-    PowerShellVersion = '5.0'
-    FunctionsToExport = @('Start-PSSwaggerLiveTestServer','Add-PSSwaggerLiveTestLibType','Add-PSSwaggerLiveTestServerType')
+    Description = 'PowerShell module with commands for building the PSSwagger.LiveTestFramework package.'
+    PowerShellVersion = '5.1'
+    FunctionsToExport = @('Initialize-Dependencies','Invoke-Build','Build-DotNetProject')
```
Comment (bmanikm) [inline PSSwagger.LiveTestFramework/build/PSSwagger.LiveTestFramework.Build.psd1]: >Build [](start = 68, length = 5)

Build is not an approved verb.

File: PSSwagger.LiveTestFramework/build/PSSwagger.LiveTestFramework.Build.psd1 [hunk 2]
```diff
@@ -1,22 +1,22 @@
...
     CmdletsToExport = ''
     VariablesToExport = ''
     AliasesToExport = ''
-    RequiredModules = @('PSSwaggerUtility')
+    RequiredModules = @()
     NestedModules = @()
-
+	DefaultCommandPrefix = 'LTFBuild'
```
Comment (bmanikm) [inline PSSwagger.LiveTestFramework/build/PSSwagger.LiveTestFramework.Build.psd1]: >LTFBuild [](start = 25, length = 8)

Consider moving this prefix to the actual functions being exported.

Also note that 'Invoke-Build' will become Invoke-LTFBuildBuild.

File: PSSwagger.LiveTestFramework/build/PSSwagger.LiveTestFramework.Build.psd1 (hunk 1 above)
Comment (bmanikm) [inline PSSwagger.LiveTestFramework/build/PSSwagger.LiveTestFramework.Build.psd1]: >Dependencies [](start = 38, length = 12)

Dependency ?

File: PSSwagger.LiveTestFramework/build/PSSwagger.LiveTestFramework.Build.psm1 [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+  Ensures all dependencies required for running tests are present on the current machine.
+#>
+function Initialize-Dependencies {
+    [CmdletBinding()]
+    param()
+
+    $expectedDotNetVersion = "2.1.0-preview1-006547"
+    Write-Host "Setting up PSSwagger.LiveTestFramework.Build dependencies:"
+    Write-Host "    dotnet: $expectedDotnetVersion"
+    Write-Host "    PSSwaggerUtility: *"
```
Comment (bmanikm) [inline PSSwagger.LiveTestFramework/build/PSSwagger.LiveTestFramework.Build.psm1]: >* [](start = 38, length = 1)

Curious to know the purpose of '*' here.

File: PSSwagger.LiveTestFramework/build/PSSwagger.LiveTestFramework.Build.psm1 [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+    Initialize-Dependencies
+    if ($OutputDirectory -and $CleanOutputDirectory -and (Test-Path -Path $OutputDirectory -PathType Container)) {
+        Remove-Item -Path $OutputDirectory -Recurse
+    }
+
+    if ($OutputDirectory -and (-not (Test-Path -Path $OutputDirectory -PathType Container))) {
+        $null = New-Item -Path $OutputDirectory -ItemType Directory
+    }
+
+    Get-ChildItem -Path (Join-Path -Path $PSScriptRoot -ChildPath ".." | Join-Path -ChildPath "src" | Join-Path -ChildPath "*.csproj") -File -Recurse | ForEach-Object {
```
Comment (bmanikm) [inline PSSwagger.LiveTestFramework/build/PSSwagger.LiveTestFramework.Build.psm1]: >Join-Path -Path $PSScriptRoot -ChildPath ".." [](start = 25, length = 45)

Consider using below for getting the parent folder path.

Split-Path -Path $PSScriptRoot -Parent

File: PSSwagger.LiveTestFramework/build/PSSwagger.LiveTestFramework.Build.psm1 [hunk 5]
```diff
@@ -0,0 +1,338 @@
...
+                Copy-Item -Path $src -Destination $OutputDirectory
+            }
+        }
+    }
+}
+
+function Build-DotNetProject {
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$false)]
```
Comment (bmanikm) [inline PSSwagger.LiveTestFramework/build/PSSwagger.LiveTestFramework.Build.psm1]: This should be mandatory, right?

File: PSSwagger.LiveTestFramework/build/PSSwagger.LiveTestFramework.Build.psm1 [hunk 6]
```diff
@@ -0,0 +1,338 @@
...
+        $Framework = 'net452',
+
+        [Parameter(Mandatory=$false)]
+        [string]
+        [ValidateSet('Debug','Release')]
+        $Configuration = 'Debug',
+
+        [Parameter(Mandatory=$false)]
+        [string]
+        $OutputDirectory,
```
Comment (bmanikm) [inline PSSwagger.LiveTestFramework/build/PSSwagger.LiveTestFramework.Build.psm1]: It looks like this param is not used in this function.

File: PSSwagger.LiveTestFramework/build/PSSwagger.LiveTestFramework.Build.psm1 [hunk 7]
```diff
@@ -0,0 +1,338 @@
...
+
+function Setup-PSSwaggerUtility {
+    [CmdletBinding()]
+    param()
+
+    if (-not (Get-Module -Name PSSwaggerUtility -ListAvailable)) {
+        $p = Find-Package -Name PSSwaggerUtility
+        if (-not $p) {
+            Write-Error -Message "Couldn't find PSSwaggerUtility package. Run 'Find-Package PSSwaggerUtility' to see error messages or recommended actions."
+            return $true
```
Comment (bmanikm) [inline PSSwagger.LiveTestFramework/build/PSSwagger.LiveTestFramework.Build.psm1]: > return $true [](start = 11, length = 13)

returning true?

File: PSSwagger.LiveTestFramework/build/PSSwagger.LiveTestFramework.Build.psm1 [hunk 8]
```diff
@@ -0,0 +1,338 @@
...
+    }
+    popd
+}
+
+function Setup-PSSwaggerUtility {
+    [CmdletBinding()]
+    param()
+
+    if (-not (Get-Module -Name PSSwaggerUtility -ListAvailable)) {
+        $p = Find-Package -Name PSSwaggerUtility
```
Comment (bmanikm) [inline PSSwagger.LiveTestFramework/build/PSSwagger.LiveTestFramework.Build.psm1]: >Find-Package -Name PSSwaggerUtility [](start = 13, length = 35)

Consider using PowerShellGet cmdlets with explicit source/repo name.
or specify the provider name and source name for *-Package cmdlets.

File: PSSwagger.LiveTestFramework/build/PSSwagger.LiveTestFramework.Build.psm1 (hunk 3 above)
Comment (brywang-msft) [inline PSSwagger.LiveTestFramework/build/PSSwagger.LiveTestFramework.Build.psm1]: * means any version should be okay

**Pull request merged**